addons/survey/models/survey.py

```diff
@@ -0,0 +1,1159 @@
+    date_create = fields.Datetime(string='Creation Date', required=True,
+                                   readonly=True, copy=False, default=fields.Datetime.now)
+    deadline = fields.Datetime(string="Deadline",
+                            help="Date by which the person can open the survey and submit answers",
+                            oldname="date_deadline")
+    input_type = fields.Selection([('manually', 'Manually'), ('link', 'Link')],
+                             string='Answer Type', required=True, readonly=True,
+                             oldname="response_type", default='manually')
+    state = fields.Selection([('new', 'Not started yet'),
+                               ('skip', 'Partially completed'),
+                               ('done', 'Completed')],
+                              string='Status',
+                              readonly=True, default='new')
+    test_entry = fields.Boolean(readonly=True)
+    token = fields.Char(string="Identification token", readonly=True, required=True, copy=False, default=lambda self: uuid.uuid4().__str__())
+
+    # Optional Identification data
+    partner_id = fields.Many2one('res.partner', string='Partner', readonly=True)
+    email = fields.Char(string="E-mail", readonly=True)
+
+    # Displaying data
+    last_displayed_page_id = fields.Many2one('survey.page',
+                                          string='Last displayed page')
+    # The answers !
+    user_input_line_ids = fields.One2many('survey.user_input_line',
+                                           'user_input_id', string='Answers', copy=True)
+
+    # URLs used to display the answers
+    result_url = fields.Char(related='survey_id.result_url',
+                                 string="Public link to the survey results")
+    print_url = fields.Char(related='survey_id.print_url',
+                                string="Public link to the empty survey")
+
+    quizz_score = fields.Float(compute="_compute_quizz_score", string="Score for the quiz")
+
+    _sql_constraints = [
+        ('unique_token', 'UNIQUE (token)', 'A token must be unique!'),
+        ('deadline_in_the_past', 'CHECK (deadline >= date_create)', 'The deadline cannot be in the past')
+    ]
+
+    @api.multi
+    def _compute_quizz_score(self):
+        for user_input in self:
+            user_input.quizz_score = sum([uil.quizz_mark for uil in user_input.user_input_line_ids])
+
+    @api.multi
+    def action_survey_resent(self):
+        ''' Sent again the invitation '''
+        self.ensure_one()
+        context = dict(self.env.context or {})
+        context.update({
+            'survey_resent_token': True,
+            'default_partner_ids': self.partner_id.id and [self.partner_id.id] or [],
+            'default_multi_email': self.email or "",
+            'default_public': 'email_private',
+        })
+        return self.survey_id.env['survey.survey'].with_context(context).action_send_survey()
```

Comment:
this statement is confused. I think, "self.with_context(context).survey_id.action_send_survey()" should be work